### PR TITLE
backports

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
ported some compression related patches + skipped some irrelevant 

skipped classes - `PublicClusterStateAction` is part of zen1 which we removed in https://github.com/crate/crate/pull/8643/commits/69de1c5a09a418d32e03b41c8051ab0886583973

Also some patches have changes in `BlobStoreRepository` which is related to cached repository data - which we ported but then reverted in https://github.com/crate/crate/commit/983c980cf6b4b4ce2acb6494dab18cb3eb0a90dd

### CODE REVIEW

File: devs/docs/es-backports.rst [hunk 1]
```diff
@@ -156,31 +157,31 @@ should be crossed out as well.
...
-- [ ] ed4984a32e7 Remove Redundant Stream Wrapping from Compression (#62017) (#62132)
+- [x] ed4984a32e7 Remove Redundant Stream Wrapping from Compression (#62017) (#62132)
 - [x] 075271758e3 Keep checkpoint file channel open across fsyncs (#61744)
 - [ ] 2bb5716b3dc Add repositories metering API (#62088)
 - [ ] bb0a583990e Allow enabling soft-deletes on restore from snapshot (#62018)
 - [ ] 3389d5ccb25 Introduce integ tests for high disk watermark (#60460)
 - [ ] 395538f5083 Improve Snapshot State Machine Performance (#62000) (#62049)
-- [ ] a295b0aa86f Fix null_value parsing for data_nanos field mapper (#61994)
-- [ ] 1799c0c5833 Convert completion, binary, boolean tests to MapperTestCase (#62004)
-- [ ] 0c8b4385777 Add support for runtime fields (#61776)
```
Comment (BaurzhanSakhariev) [inline devs/docs/es-backports.rst:167]: `FieldMapper` patches

File: devs/docs/es-backports.rst [hunk 2]
```diff
@@ -156,31 +157,31 @@ should be crossed out as well.
...
+- [s] a295b0aa86f Fix null_value parsing for data_nanos field mapper (#61994)
+- [s] 1799c0c5833 Convert completion, binary, boolean tests to MapperTestCase (#62004)
+- [s] 0c8b4385777 Add support for runtime fields (#61776)
 - [ ] b26584dff89 Remove unused deciders in BalancedShardsAllocator (#62026)
-- [ ] 6d08b55d4e3 Simplify searchable snapshot shard allocation (#61911)
-- [ ] 66bb1eea982 Improve error messages on bad [format] and [null_value] params for date mapper (#61932)
-- [ ] 31c026f25cc upgrade to Lucene-8.7.0-snapshot-61ea26a (#61957) (#61974)
-- [ ] af01ccee93e Add specific test for serializing all mapping parameter values (#61844) (#61877)
-- [ ] d59343b4ba8 Allow [null] values in [null_value] (#61798) (#61807)
```
Comment (BaurzhanSakhariev) [inline devs/docs/es-backports.rst:173]: SearchableSnapshot and FieldMapper related patches

File: devs/docs/es-backports.rst [hunk 3]
```diff
@@ -156,31 +157,31 @@ should be crossed out as well.
...
+- [s] 6d08b55d4e3 Simplify searchable snapshot shard allocation (#61911)
+- [s] 66bb1eea982 Improve error messages on bad [format] and [null_value] params for date mapper (#61932)
+- [s] 31c026f25cc upgrade to Lucene-8.7.0-snapshot-61ea26a (#61957) (#61974)
+- [s] af01ccee93e Add specific test for serializing all mapping parameter values (#61844) (#61877)
+- [s] d59343b4ba8 Allow [null] values in [null_value] (#61798) (#61807)
 - [ ] 3fd25bfa877 Fix Concurrent Snapshot Create+Delete + Delete Index (#61770) (#61773)
 - [s] 5723b928d7d Remove Outdated Snapshot Docs (#61684) (#61728)
-- [ ] 1bfebd54ea7 [7.x] Allocate newly created indices on data_hot tier nodes (#61342) (#61650)
+- [s] 1bfebd54ea7 [7.x] Allocate newly created indices on data_hot tier nodes (#61342) (#61650)
```
Comment (BaurzhanSakhariev) [inline devs/docs/es-backports.rst:177]: AFAIK `data_hot` tier is irrlevenat thing for us

File: devs/docs/es-backports.rst [hunk 4]
```diff
@@ -156,31 +157,31 @@ should be crossed out as well.
...
 - [s] f769821bc80 Pass SearchLookup supplier through to fielddataBuilder (#61430) (#61638)
 - [ ] b866aaf81c0 Use int for number of parts in blob store (#61618)
-- [ ] 9f566644af2 Do not create two loggers for DeprecationLogger backport(#58435)  (#61530)
```
Comment (BaurzhanSakhariev) [inline devs/docs/es-backports.rst:179]: Skipped `DeprecationLogger` related patches, we did a lot of change there in https://github.com/crate/crate/commit/7ac48662cc8015b6ee14f063ca67daccc72c1bda

**Review by matriv — APPROVED** [review]
LGTM!

@mfussenegger Could you please double check the skipped patches?
@BaurzhanSakhariev very minor: https://github.com/crate/crate/pull/13358/commits/993deb2ed6ec873449a5488f315a481f1982d052 typo `ski`->`skip` in commit message

**Review by mfussenegger — APPROVED** [review]

**Comment by BaurzhanSakhariev** [comment]
> FYI, @BaurzhanSakhariev fixed typo in commit message myself

thanks!

**Pull request merged**